media-source-respec.html

```diff
@@ -1155,13 +1260,30 @@ <h2><dfn>SourceBuffer</dfn> Object</h2>
                     attribute EventHandler        onerror;
                     attribute EventHandler        onabort;
     undefined appendBuffer (BufferSource data);
+    Promise < undefined > appendEncodedChunks(EncodedChunks chunks);
     undefined abort ();
-    undefined changeType (DOMString type);
+
+    // "optional" with default empty SourceBufferConfig dictionary is used here
+    // to pass WebIDL verification. Behavior of the method enforces that either
+    // a valid mime-type string or a valid SourceBufferConfig are provided.
+    undefined changeType (optional TypeOrConfig typeOrConfig = {});
+
     undefined remove (double start, unrestricted double end);
-};</pre>
+};
+
+// This typedef simplifies the syntax for describing either a single audio or
+// video chunk, or a sequence of chunks. While this typedef does not provide the
+// ability to enforce that all members of a sequence of chunks are the same kind
+// (audio versus video), the behavior of the appendEncodedChunks method enforces
+// that constraint: the implementation MUST reject a sequence containing both
+// audio and video chunks.
+typedef (sequence< (EncodedAudioChunk or EncodedVideoChunk) >
+            or EncodedAudioChunk or EncodedVideoChunk) EncodedChunks;
```

Comment:
Can't we write `sequence<EncodedAudioChunk> or sequence<EncodedVideoChunk> or EncodedAudioChunk or EncodedVideoChunk` ?

Reply:
I believe I tried that originally in the prototype, but the IDL bindings generator failed. I fuzzily recall maybe it was due to an identifier length limit on the generated type name. Worth reconsidering.

Reply:
I don't like leaving this with just a fuzzy recollection, so I checked the [Chromium prototype's commit description](https://source.chromium.org/chromium/chromium/src/+/6507c9cc4ae2d08d090d466da71741d8677380cf) for the change I landed that added this IDL:

>   Caveat: "sequence<A> or sequence<V>" cannot be disambiguated by the
>   bindings, so sequence<A or V> is used in this change. Regardless, the
>   eventual implementation would need to validate that all in the
>   sequence are either A or all are V (along with the usual validation
>   that appended chunks or frames also appear to use the most recent
>   SourceBufferConfig).

Likewise, my fuzzy recollection of identifier length was fuzzy. In that same change I *did* run into identifier length limits, but solved that by adding the long identifier into an exception list for that check in the generator:

>   Caveat: The bindings generator requires help when generated union type
>   identifiers are too long for some platforms. This change adds a
>   seventh case to the existing hard-coded lists of names that need
>   shortening with the generator.